src/storr.c

```diff
@@ -0,0 +1,34 @@
+#include <R.h>
+#include <R_ext/Rdynload.h>
+#include <Rinternals.h>
+#include <Rversion.h>
+
+SEXP Cread_text_file(SEXP path, SEXP nchar) {
+  FILE *fp;
+  fp = fopen(CHAR(asChar(path)), "rb");
+  if (fp == NULL) {
+    return R_NilValue;
+  }
+  int n = asInteger(nchar) + 1; // Need an extra character for '\0'.
+  char *buf = (char*) malloc(n * sizeof(char));
```

Comment:
If used a stack allocated array this would be a bit simpler. In general one should use `R_alloc()` here otherwise as R will clean that up.

So something like

```
#define MAX_HASH_LENGTH 128
...

char buf[MAX_HASH_LENGTH + 1];
char * res = fgets(buf, n, fp);
if (res == NULL) {
  Rf_error("empty file"); // please test!
}
fclose(fp);
```

then the `free` can come out too

Reply:
Maybe faster too.

Reply:
73fb9236f128a65d24f839e8d021378cd483c0ee